### PR TITLE
Scripts/Spells: Add script for Salvage Wreckage & update similar scripts

### LINKED ISSUE #25099: DB/Quest: Recover the Cargo!
**Description:**

When using the **Salvage Kit** on **Shipwreck Debris**, nothing happens. No **Salvaged Strongbox** or **Mirefin Ambusher** appears.
There was a similar [issue](https://github.com/TrinityCore/TrinityCore/issues/22245) before.

**Expected behaviour:**

According to [Wowhead](https://www.wowhead.com/quest=11140/deprecated-recover-the-cargo#comments) comments, the item or the ambusher should appear.

**Steps to reproduce the problem:**

1. `.go creature id 23569`
2. `.quest add 11140`
3. Do the quest normally and try to salvage the **Salvaged Strongboxes**.

**Branch(es):**

3.3.5

**TC rev. hash/commit:** 

d974800e9d1e393ce3459ed8be4f4bd9defcc8f3

**Operating system:** Windows 7 Enterprise 64bit

### PR DESCRIPTION
**Pull request by offl**
**Changes proposed:**

Remove one spell script because Activate Object now can and should be handled in core
Remove Activate Object effect from other 2 scripts and few more changes(modern standarts, validate spells; in one of those scripts use spell to create item instead of direct AddItem(because of that we don't need anymore to check if caster is player), missing conditions to make everything work)
Migrate effect of spell Salvage Wreckage from SAI to spell script in order to fix spell effect. SAI script stopped\never worked (#25053) because gameobject simply despawns before event execution

**Target branch(es):** 3.3.5/master

- [x] 3.3.5
- [ ] master

**Issues addressed:**

Closes #25099

**Tests performed:**

Builds, tested affected spells, works

**Known issues and TODO list:** (add/remove lines as needed)

- [ ] spell_q14076_14092_pound_drum despawns object not instantly before and after changes. I was told 'there was sth in 335 that delays despawns in general'. It's not a TO-DO, just a note for future devs

### CODE REVIEW

File: src/server/scripts/Spells/spell_quest.cpp [hunk 1]
```diff
@@ -1112,125 +1112,68 @@ class spell_q14112_14145_chum_the_water: public SpellScriptLoader
...
-    public:
-        spell_q9452_cast_net() : SpellScriptLoader("spell_q9452_cast_net") { }
-
-        class spell_q9452_cast_net_SpellScript : public SpellScript
-        {
-            PrepareSpellScript(spell_q9452_cast_net_SpellScript);
-
-            bool Load() override
-            {
-                return GetCaster()->GetTypeId() == TYPEID_PLAYER;
```
Comment (ccrs) [inline src/server/scripts/Spells/spell_quest.cpp:1134]: reasoning about this change?

File: src/server/scripts/Spells/spell_quest.cpp [hunk 2]
```diff
@@ -1689,6 +1632,38 @@ class spell_q11010_q11102_q11023_q11008_check_fly_mount : public SpellScriptLoad
...
+    {
+        return ValidateSpellInfo({ SPELL_SUMMON_LOCKBOX, SPELL_SUMMON_BURROWER });
+    }
+
+    void HandleDummy(SpellEffIndex /*effIndex*/)
+    {
+        Unit* caster = GetCaster();
+
+        if (roll_chance_i(50))
+                caster->CastSpell(caster, SPELL_SUMMON_LOCKBOX, true);
```
Comment (ccrs) [inline src/server/scripts/Spells/spell_quest.cpp:1656]: minor codestyle here

Comment (offl) [reply src/server/scripts/Spells/spell_quest.cpp]: Ty

**Comment by offl** [comment]
Looks like this was reviewed

**Pull request merged**